src/shared/api/dto.ts

```diff
@@ -1,183 +1,154 @@
-     */
-    timestamp: string
-    block_hash: string
-    /**
-     * Transactions count
-     */
-    transactions: number
-    /**
-     * Rejected transactions count
-     */
-    rejected_transactions: number
-  }
-
-  export interface Block {
-    height: number
-    /**
-     * See {@link BlockShallow.timestamp}
-     */
-    timestamp: string
-    block_hash: string
-    parent_block_hash: string
-    rejected_transactions_merkle_root_hash: string
-    invalidated_blocks_hashes: string[]
-    /**
-     * List of serialized {@link @iroha2/data-model#VersionedValidTransaction}
-     */
-    transactions: string[]
-    /**
-     * List of serialized {@link @iroha2/data-model#VersionedRejectedTransaction}
-     */
-    rejected_transactions: string[]
-    /**
-     * List of hashes. WIP always empty
-     */
-    view_change_proofs: string[]
-  }
-
-  export interface TransactionDto {
-    /**
-     * WIP zeroed
-     */
-    hash: string
-    block_hash: string
-    // TODO: bloch_height is missing from backend response
-    block_height: number
-    payload: TransactionPayload
-    signatures: Signature[]
-    rejection_reason?: string
-  }
-
-  export interface TransactionPayload {
-    account_id: string
-    instructions: TransactionInstructions
-    /**
-     * ISO timestamp
-     */
-    creation_time: string
-    time_to_live_ms: number | null
-    nonce: number | null
-    metadata: any
-  }
-
-  /**
-   * `Instructions` `string[]` - list of serialized
-   * {@link @iroha2/data-model#Instruction}
-   */
-  export type TransactionInstructions = Tagged<'Instructions', string[]> | Tagged<'Wasm', undefined>;
-
-  export interface Signature {
-    /**
-     * Public key's multihash
-     */
-    public_key: string
-    /**
-     * Hex binary
-     */
-    payload: string
-  }
+import { z } from 'zod';
+import BigNumber from 'bignumber.js';
+
+const paginationSchema = z.object({
+  page: z.number(),
+  per_page: z.number(),
+  // TODO: remove nullable when backend is ready
+  total_pages: z.number().nullable(),
+  total_items: z.number().nullable(),
+});
+
+export type Pagination = z.infer<typeof paginationSchema>;
+
+export interface Paginated<T> {
+  pagination: Pagination
+  items: T[]
 }
+
+const paginationParamsSchema = z
+  .object({
+    page: z.number(),
+    per_page: z.number(),
+  })
+  .partial();
+
+export type PaginationParams = z.infer<typeof paginationParamsSchema>;
+
+const metadataSchema = z.record(z.string(), z.any());
+
+const accountIdSchema = z.string().brand('AccountId');
+const domainIdSchema = z.string().brand('DomainId');
+const assetDefinitionIdSchema = z.string().brand('AssetDefinitionId');
+const assetIdSchema = z.string().brand('AssetId');
+
+export const accountSchema = z.object({
+  id: accountIdSchema,
+  metadata: metadataSchema,
+});
+
+export type Account = z.infer<typeof accountSchema>;
+
+const assetSearchParamsSchema = paginationParamsSchema.extend({
+  owned_by: accountIdSchema.optional(),
+});
```

Comment:
This is a type that doesn't need validation/transformation, therefore, schema looks needless here.